src/Symfony/Component/Console/Command/InvokableCommand.php

```
@@ -188,4 +192,61 @@ public function handleSignal(int $signal, int|false $previousExitCode = 0): int|
     {
         return $this->signalableCommand?->handleSignal($signal, $previousExitCode) ?? false;
     }
+
+    public function isInteractive(): bool
+    {
+        if (null === $this->interactions) {
+            $this->collectInteractions();
+        }
+
+        return [] !== $this->interactions;
+    }
+
+    public function interact(InputInterface $input, OutputInterface $output): void
+    {
+        if (null === $this->interactions) {
+            $this->collectInteractions();
+        }
+
+        foreach ($this->interactions as $interaction) {
+            $interaction->interact($input, $output, $this->getParameters(...));
+        }
+    }
+
+    private function collectInteractions(): void
+    {
+        $invokableThis = $this->invokable->getClosureThis();
+
+        $this->interactions = [];
+        foreach ($this->invokable->getParameters() as $parameter) {
+            if ($spec = Argument::tryFrom($parameter)) {
+                if ($interactive = $spec->getInteractive()) {
+                    $this->interactions[] = new Interaction($invokableThis, $interactive);
+                }
+
+                continue;
+            }
+
+            if ($spec = Input::tryFrom($parameter)) {
+                foreach ($spec->getPropertyInteractions() as $interaction) {
+                    $this->interactions[] = $interaction;
+                }
+
+                foreach ($spec->getMethodInteractions() as $interaction) {
+                    $this->interactions[] = $interaction;
+                }
+            }
+        }
+
+        if (!$class = $this->invokable->getClosureCalledClass()) {
+            return;
+        }
+
+        foreach ($class->getMethods() as $method) {
```

Why collecting things from all class methods ?

---

It's not. In `Interactive::tryFrom()`, we first check whether the attribute is attached, then verify if it's public and non-static, so that we throw a logic exception instead of ignoring it.

---

Well, I just excluded the `__invoke` method as an allowed interactive method, since that could be confusing

---

I agree this makes sense.